include/xtensor/xbuilder.hpp

```diff
@@ -138,7 +138,8 @@ namespace xt
     template <class E>
     inline typename E::temporary_type empty_like(const xexpression<E>& e)
     {
-        typename E::temporary_type res(e.derived_cast().shape());
+        using xtype = typename E::temporary_type;
+        auto res = xtype::from_shape(e.derived_cast().shape());
         return res;
```

Comment:
What is the difference? `from_shape` appears to be calling the constructor of `temporary_type. Is this meant to cover cases where `e.shape` could not be used to construct the shape type of the temporary type?

Reply:
yes, that's it. for example the shape types of the new xview's will be sequence_view. calling the `from_shape` guarantees a construction a copy from iterators.

Reply:
+1